lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -896,6 +896,179 @@ class ConvertAtenUpsampleNearest2dVecOp
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: the scale_factors is not constructed from "
+                "ListConstruct");
+      scaleFactorsFloatValues = getTypeConvertedValues(
+          rewriter, loc, getTypeConverter(), scaleFactorsTorchFloat);
+      for (unsigned i = 0; i < scaleFactorsFloatValues.size(); i++)
+        scaleFactorsFloatValues[i] =
+            convertScalarToDtype(rewriter, loc, scaleFactorsFloatValues[i],
+                                 mlir::Float32Type::get(op->getContext()));
+    } else {
+      for (unsigned i = 0; i < 2; i++) {
+        auto scaleFactorVal = rewriter.create<arith::DivFOp>(
+            loc,
+            convertScalarToDtype(rewriter, loc,
+                                 gradOutputSizeIntValues[hDimOffset + i],
+                                 mlir::Float32Type::get(op->getContext())),
+            convertScalarToDtype(rewriter, loc,
+                                 inputSizeIntValues[hDimOffset + i],
+                                 mlir::Float32Type::get(op->getContext())));
+        scaleFactorsFloatValues.push_back(scaleFactorVal);
+      }
+    }
+
+    SmallVector<Value, 2> scaleFactorsIntValues;
+    for (auto v : scaleFactorsFloatValues)
+      scaleFactorsIntValues.push_back(convertScalarToDtype(
+          rewriter, loc, rewriter.create<math::CeilOp>(loc, v),
+          mlir::IntegerType::get(op->getContext(), 64)));
+
+    Value outTensor = createZeroInitTensor(
+        rewriter, loc,
+        castIntVectorToIndexVector(rewriter, loc, inputSizeIntValues),
+        elementType);
+
+    Value kernelTensor = rewriter.create<tensor::EmptyOp>(
+        loc,
+        getAsOpFoldResult(
+            castIntVectorToIndexVector(rewriter, loc, scaleFactorsIntValues)),
+        elementType);
+
+    SmallVector<AffineExpr> affineExprs;
+    for (unsigned i = 0; i < gradOutputRank; i++)
+      affineExprs.push_back(rewriter.getAffineDimExpr(i));
+
+    AffineMap inputMap = AffineMap::get(gradOutputRank + 2, /*symbolCount=*/0,
+                                        affineExprs, op->getContext());
+
+    affineExprs.clear();
+    for (unsigned i = 0; i < 2; i++)
+      affineExprs.push_back(rewriter.getAffineDimExpr(gradOutputRank + i));
+
+    AffineMap kernelMap = AffineMap::get(gradOutputRank + 2, /*symbolCount=*/0,
+                                         affineExprs, op->getContext());
+
+    SmallVector<AffineMap> indexingMaps{inputMap, kernelMap, inputMap};
+    SmallVector<StringRef> iteratorTypes(gradOutputRank,
+                                         getParallelIteratorTypeName());
+    iteratorTypes.push_back(getReductionIteratorTypeName());
+    iteratorTypes.push_back(getReductionIteratorTypeName());
+
+    Value finalRes =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, outTensor.getType(), ValueRange{outTensor, kernelTensor},
+                ValueRange{outTensor},
```

Comment:
I don't think you need to pass `outTensor` as an input. You have access to `outTensor` as the output of the linalg::GenericOp

Reply:
In that case, I need to create another dummy tensor which would be passed as an input. Because the iteration is over the `outTensor`, which gives these indices: https://github.com/vivekkhandelwal1/torch-mlir/blob/0c388c18e1a9260811d53c8d957a84661a876f5d/lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp#L1051-L1054

Reply:
I'm not sure I understand. You should be able to get the indices for the `outTensor` without needing to pass `outTensor` as an input. The indices are generated using the information from the `AffineMap`s, which in this case have a domain of 6 dimensions. Since the output tensor of the `linalg::GenericOp` uses the `inputMap` as it's indexing map, then the indices should work without having to pass `outTensor` as an input.

Reply:
Thanks, I figured this out.